docs/blog/2019-01-04-seo-and-gatsby/index.md

```diff
@@ -0,0 +1,449 @@
+---
+title: "Search Engine Optimization with Gatsby"
+date: 2019-01-04
+author: Dustin Schau
+cover: images/seo.jpg
+excerpt: "SEO and Gatsby: A Perfect Pairing. Learn how Gatsby implements SEO utilizing React Helmet and smart defaults and how you can use these tools to implement your own!"
+tags:
+  - gatsby
+  - javascript
+  - react
+  - seo
+---
+
+Search Engine Optimization (hereafter SEO) is something that you should want. You've possibly even been approached by an SEO _expert_ who can maximize your revenue and page views just by following these **Three Simple Tricks**! However, relatively few make the concerted effort to implement SEO in their web app. In this post, I'll share some of the ins and outs of SEO and how you can implement common, simple SEO patterns in your Gatsby web app, today. By the end of this post you'll know how to do the following:
+
+- Implement SEO patterns with [react-helmet][react-helmet]
+- Create an optimized social sharing card for Twitter, Facebook, and Slack
+- Tweak the SEO component exposed in the default gatsby starter ([`gatsby-starter-default`][gatsby-starter-default])
+
+## Implementation
+
+The core technology powering SEO is the humble, ubiquitiuous `meta` tag. You've probably seen something like:
+
+```html
+<meta
+  name="viewport"
+  content="width=device-width, initial-scale=1, shrink-to-fit=no"
+/>
+```
+
+or further still with more of an SEO spin something as simple as the `description` and/or `keywords` properties:
+
+```html
+<meta
+  name="description"
+  content="This is probably some earth-shattering excerpt that is around ~200 characters or less"
+/>
+<meta
+  name="keywords"
+  content="science dog, another term, do-i-use-spaces, hello"
+/>
+```
+
+These are the _bare minimum_ requirements that should be implemented for simple and basic SEO. However--we can go further, and we can go further with the powerful combo of content rendered at _build time_ powered by Gatsby and GraphQL. Let's dive in.
+
+## Gatsby + GraphQL
+
+GraphQL is a crucial feature enabled via Gatsby (note: you don't [_have_ to use GraphQL with Gatsby][unstructured-data]). Leveraging GraphQL to query your indexable content--wherever it lives (at build time!)--is one of the most powerful and flexible techniques enabled via Gatsby. Let's briefly look at how we can implement an extensible and flexible SEO component.
+
+### `StaticQuery`
+
+Gatsby distinguishes between page-level queries and component queries. The former can use page GraphQL queries while the latter can use a new in Gatsby v2 feature called [`StaticQuery`][gatsby-static-query]. A StaticQuery will be parsed, evaluated, and injected at _build time_ into the component that is requesting the data. This is a perfect scenario in which we can create an SEO component with sane defaults that can be easily extended.
+
+### Creating the component
```

Comment:
I think this header and first sentence could be more specific for easier scanning